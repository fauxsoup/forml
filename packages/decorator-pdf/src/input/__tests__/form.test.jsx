import Form from "../form";
import Context from "@forml/context";
import React from "react";
import ReactPDF from "@react-pdf/renderer";
import { render } from "@testing-library/react";
import * as decorator from "../../";

describe("renders", function() {
    let form;
    let styles;
    let input;
    let value;

    beforeEach(function() {
        form = { type: "string" };

        input = jest.fn(() => ({ border: "1pt solid red" }));

        styles = {
            get input() {
                return input();
            },
        };
    });

    test("with relevant formStyles", function() {
        form = { ...form, styles };
        const { container } = render(
            <ReactPDF.Page>
                <Context.Provider value={{ decorator }}>
                    <Form form={form} value={value} />
                </Context.Provider>
            </ReactPDF.Page>
        );

        expect(container).toMatchSnapshot();
        expect(input).toHaveBeenCalled();
    });
});
