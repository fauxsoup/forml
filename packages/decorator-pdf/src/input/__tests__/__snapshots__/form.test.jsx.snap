// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with relevant formStyles 1`] = `
<div>
  <page>
    <view
      style="min-height: 14pt; font-family: Roboto; font-size: 10pt; flex: 1; padding: 2pt; border: 1pt solid red;"
    >
      <text />
    </view>
  </page>
</div>
`;
